Agents/BuildingIdentificationAgent/BuildingIdentificationAgent/src/main/java/uk/ac/cam/cares/jps/agent/buildingidentification/BuildingIdentificationAgent.java

```diff
@@ -0,0 +1,541 @@
+package uk.ac.cam.cares.jps.agent.buildingidentification;
+
+import javax.servlet.annotation.WebServlet;
+import javax.ws.rs.BadRequestException;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.core.io.ClassPathResource;
+import com.cmclinnovations.stack.clients.ontop.OntopClient;
+
+import uk.ac.cam.cares.jps.base.agent.JPSAgent;
+import uk.ac.cam.cares.jps.base.query.RemoteRDBStoreClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.DSLContext;
+import org.jooq.SQLDialect;
+import org.jooq.exception.DataAccessException;
+import org.jooq.impl.DSL;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.*;
+import java.util.*;
+import uk.ac.cam.cares.jps.base.util.CRSTransformer;
+
+@WebServlet(urlPatterns = { BuildingIdentificationAgent.ROUTE_LOCATION,
+        BuildingIdentificationAgent.ROUTE_POSTGIS
+})
+public class BuildingIdentificationAgent extends JPSAgent {
+
+    public static final String KEY_REQ_URL = "requestUrl";
+    public static final String ROUTE_LOCATION = "/location";
+    public static final String ROUTE_POSTGIS = "/postgis";
+    public static final String KEY_DISTANCE = "maxDistance";
+    public static final String KEY_COORD = "coordinates";
+    public static final String KEY_SRID = "srid";
+    public static final String KEY_TABLE = "table";
+    public static final String KEY_COLUMN = "column";
+    public static final String KEY_ONE_MANY = "oneToMany";
+    public static final String KEY_NEW_TABLE = "newTable";
+    public static final String KEY_FILTER_COLUMNS = "filterColumns";
+    public static final String KEY_EXCLUDED_VALUES = "excludedValues";
+    public static final String KEY_OVERLAP_FRACTION = "overlapFraction";
+    public static final String KEY_MAPPING = "mapping";
+    private static final String EPSG = "EPSG:";
+    public static final String POINT_TYPE = "ST_Point";
+
+    private static final Logger LOGGER = LogManager.getLogger(BuildingIdentificationAgent.class);
+
+    private static final String BUILDINGS_COLUMN = "building_uuid";
+
+    // RDBStore client to query postgis
+    private RemoteRDBStoreClient rdbStoreClient;
+
+    @Override
+    public JSONObject processRequestParameters(JSONObject requestParams) {
+        if (validateInput(requestParams)) {
+
+            // Properties of database containing buildings data.
+            String dbUrl = null;
+            String dbUser = null;
+            String dbPassword = null;
+
+            if (requestParams.has("dbUrl")) {
+                dbUrl = requestParams.getString("dbUrl");
+                dbUser = requestParams.getString("dbUser");
+                dbPassword = requestParams.getString("dbPassword");
```

Comment:
I hope this is just for testing? Passing passwords in HTTP requests is not a good idea

Reply:
I have removed this part of the code, which was initially included for testing purposes.